### PR TITLE
Migrate callbacks to promises

### PR DESCRIPTION
**Pull request by HALLERPierre**
## Todo

- [x] Replace all functions with callbacks by promises

### CODE REVIEW

File: examples/basic/basic.js [hunk 1]
```diff
@@ -36,4 +34,4 @@ horde.mogwai(horde.mogwais.fps());
 horde.mogwai(horde.mogwais.gizmo().maxErrors(2));
 
 // Unleash the horde
-horde.unleash({ nb: 1000, delay: 1 });
+horde.unleash({ nb: 200, delay: 10 });
```
Comment (zyhou) [inline examples/basic/basic.js]: Why do you change this value?

File: src/utils/executeInSeries.js [hunk 2]
```diff
@@ -9,19 +9,12 @@
...
  * @param {Function} done - The final callback to execute once all functions are executed
  */
-export default (callbacks, args, context, done) => {
-    const cbs = [...callbacks]; // clone the array to avoid modifying the original
-
-    const iterator = (iteratorCallbacks, iteratorArgs) => {
-        if (!iteratorCallbacks.length) {
-            return typeof done === 'function' ? done() : true;
+export default (callables, args, context) =>
+    new Promise(async (resolve, reject) => {
```
Comment (zyhou) [inline src/utils/executeInSeries.js]: I think we can do better with Promise.allSettled/Promise.race (https://2ality.com/2019/08/promise-combinators.html). Maybe Promise.any (https://2ality.com/2019/12/promise-any.html)

We can keep this way and improve after the general review

File: src/strategies/bySpecies.js [hunk 3]
```diff
@@ -42,47 +43,31 @@ export default () => {
...
     let stopped;
-    let doneCallback;
 
-    const bySpeciesStrategy = (newGremlins, params, done) => {
+    const bySpeciesStrategy = async (newGremlins, params) => {
         const nb = params && params.nb ? params.nb : config.nb;
-        const gremlins = newGremlins.slice(0); // clone the array to avoid modifying the original
+        const delay = params && params.delay ? params.delay : config.delay;
+
+        const gremlins = [...newGremlins]; // clone the array to avoid modifying the original
```
Comment (zyhou) [inline src/strategies/bySpecies.js]: The comment :D 

Need to review somes comment on next PR

File: examples/basic/basic.js (hunk 1 above)
Comment (HALLERPierre) [reply examples/basic/basic.js]: 1ms delay is too short and 1000 gremlins is a lot.

**Pull request merged**